tests/vip-integrations/test-vip-integrations.php

```diff
@@ -3,52 +3,85 @@
 namespace Automattic\VIP\Integrations;
 
 use WP_UnitTestCase;
+use InvalidArgumentException;
 
 require_once __DIR__ . '/fake-integration.php';
 
-/**
- * @runTestsInSeparateProcesses
- * @preserveGlobalState disabled
- */
 class VIP_Integrations_Test extends WP_UnitTestCase {
 	public function setUp(): void {
-		// Remove existing default integrations to avoid conflicts when re-registering
-		Integrations::instance()->remove_registered();
+		parent::setUp();
+
+		// Reset static is_loaded status on integration so it can be reloaded
+		FakeIntegration::$is_loaded = false;
 	}
 
-	public function test_register_integration_as_class_name() {
-		add_action( 'register_vip_integrations', function() {
-			Integrations::instance()->register( 'fake-integration', FakeIntegration::class );
-		});
+	// Registration
+
+	public function test_register_integration_as_class_name_loads() {
+		$integrations = new Integrations();
+		$integrations->register( 'fake-integration-class', FakeIntegration::class );
 
-		activate( 'fake-integration' );
+		$integrations->activate( 'fake-integration-class' );
 
-		$this->run_integration_actions();
-		$this->assertTrue( FakeIntegration::$is_integrated );
+		$integrations->load_active();
+		$this->assertTrue( FakeIntegration::$is_loaded );
 	}
 
-	public function test_register_integration_as_instantiated_class() {
+	public function test_register_integration_as_instantiated_class_loads() {
 		$fake_integration = new class() extends Integration {
```

Comment:
Why are you repeating the definition of FakeIntegration as an anonymous class? Just register an object of the FakeIntegrations class.

```php
// Create the collection.
$integrations = new Integrations();
// Register an instantiated object.
$integrations->register( 'fake-integration-instance', new FakeIntegration() );
```

The only thing that should be different in this test compared to the previous one, is the use of `new FakeIntegration()` instead of `FakeIntegration::class`.